spec/defines/attr_spec.rb

```diff
@@ -5,28 +5,86 @@
 
   let(:title) { 'test' }
 
-  let(:params) do
-    {
-      source: 'puppet:///modules/test/path/to/file',
-    }
+  context 'No specific relaxed value' do
+    let(:params) do
+      {
+        source: 'puppet:///modules/test/path/to/file',
+      }
+    end
+
+    it do
+      is_expected.to contain_file('/etc/raddb/mods-config/attr_filter/test')
+        .that_notifies('Service[radiusd]')
+        .that_requires('Group[radiusd]')
+        .that_requires('Package[freeradius]')
+        .with_ensure('present')
+        .with_group('radiusd')
+        .with_mode('0640')
+        .with_owner('root')
+        .with_source('puppet:///modules/test/path/to/file')
+    end
+
+    it do
+      is_expected.to contain_concat__fragment('attr-test')
+        .with_content(%r{^attr_filter filter.test {\n\s+key = "\%{User-Name}"\n\s+filename = \${modconfdir}/\${\.:name}/test\n}})
+        .with_order('20')
+        .with_target('/etc/raddb/mods-available/attr_filter')
+    end
   end
 
-  it do
-    is_expected.to contain_file('/etc/raddb/mods-config/attr_filter/test')
-      .that_notifies('Service[radiusd]')
-      .that_requires('Group[radiusd]')
-      .that_requires('Package[freeradius]')
-      .with_ensure('present')
-      .with_group('radiusd')
-      .with_mode('0640')
-      .with_owner('root')
-      .with_source('puppet:///modules/test/path/to/file')
+  context 'relaxed = no' do
+    let(:params) do
+      {
+        source: 'puppet:///modules/test/path/to/file',
+        relaxed: 'no',
+      }
+    end
+
+    it do
+      is_expected.to contain_file('/etc/raddb/mods-config/attr_filter/test')
+        .that_notifies('Service[radiusd]')
+        .that_requires('Group[radiusd]')
+        .that_requires('Package[freeradius]')
+        .with_ensure('present')
+        .with_group('radiusd')
+        .with_mode('0640')
+        .with_owner('root')
+        .with_source('puppet:///modules/test/path/to/file')
+    end
+
+    it do
+      is_expected.to contain_concat__fragment('attr-test')
+        .with_content(%r{^attr_filter filter.test {\n\s+key = "\%{User-Name}"\n\s+filename = \${modconfdir}/\${\.:name}/test\n\s+relaxed = no\n}})
+        .with_order('20')
+        .with_target('/etc/raddb/mods-available/attr_filter')
+    end
   end
```

Comment:
- within each relaxed* context, we can test only for the change in the relaxed value. We can use super().merge() to pull in the parent context's params:
```suggestion
  context 'with relaxed = no' do
    let(:params) do
      super().merge({ relaxed: 'no' })
    end

    it do
      is_expected.to contain_concat__fragment('attr-test')
        .with_content(%r{^\s+relaxed\s+=\s+no$}})
    end
  end
```